Project 27 - Update MARS documentation (#30)
elixir-europe/biohackathon-projects-2023: 27/mars_documentation.md

## Components in MARS
MARS is comprised of distinct components, each representing key roles in the data submission process. Together, they form the essential building blocks of MARS, enabling cohesive and reliable data submission to multiple repositories.
Each role performs specific functions and carries distinct responsibilities essential for the smooth operation of the system.
### End-User
This individual is the user who inputs experimental metadata and location of related data files into ISA-JSON Producer platforms. End-User provides essential experimental details that are encoded into ISA-JSON format.

End-User could be also direct user of the Data Broker platform, particularly in scenarios where the broker platform is only accessible as a web service within the institute's intranet.

Main function: 
* Input experimental metadata into ISA-JSON Producer platforms following the specific rules, structures, and requirements of the respective platform. 
* Use the ISA-JSON producer platfom to expor experimental metadata in ISA-JSON files and uploading them to the Data Broker platform.

> Responsibilities:

Reviewer: ```suggestion

Responsibilities:
```

Each role performs specific functions and carries distinct responsibilities essential for the smooth operation of the system.

> 

Reviewer: ```suggestion
### End-User
This individual is the user who inputs experimental metadata and location of related data files into ISA-JSON Producer platforms. End-User provides essential experimental details that are encoded into ISA-JSON format.

End-User could be also direct user of the Data Broker platform, particularly in scenarios where the broker platform is only accessible as a web service within the institute's intranet.

Main function: 
* Input experimental metadata into ISA-JSON Producer platforms following the specific rules, structures, and requirements of the respective platform. 
* Use the ISA-JSON producer platfom to expor experimental metadata in ISA-JSON files and uploading them to the Data Broker platform.
Responsibilities:
* End-User ensures the accuracy and completeness of metadata.
* End-User must refrain from uploading identical ISA-JSON files to the Data Broker platform, unless it is to initiate a distinct action such as update or release.
```

### ISA-JSON producers
This role includes any source of ISA-JSON files containing the metadata of multi-omics research. For example, research infrastructure or service facilities that operate platforms such as ARC, FAIRDOM-SEEK, ISA Creators. 

* Main function: Prepare research metadata in ISA-JSON format (i.e. generate and export ISA-JSON files)
* Responsibilities:

>   * Avoid data duplication. For example, not re-submitting already existing data in the target repositories through MARS.

Reviewer: ```suggestion
```

  * Ensure compliance of the submitted ISA-JSON with the ISA model expected by MARS.

>   * Conform submitted ISA-JSON to target repositories' additional validation rules validated by MARS. This includes repositories’ checklists, controlled vocabulary lists, ontology terms, etcetera.

Reviewer: Is MARS going to do the validation of checklists etc? I thought that would be done by target archives during submission?

>   * Conform submitted ISA-JSON to target repositories' additional validation rules validated by MARS. This includes repositories’ checklists, controlled vocabulary lists, ontology terms, etcetera.

Reviewer: ```suggestion
  * Validate ISA-JSON structure and fields in ISA-JSON as expected by MARS (e.g. “Comment[Target repository]” at Assay level). This includes verification of the checksums associated to the data files. This step might also include validation of repositories’ checklists, controlled vocabulary lists, ontology terms etc, if validation rules are provided by each repository for MARS.
```

>   * Assert that files referenced within the ISA-JSON metadata are uploaded to the target repositories. For example, if a FASTQ file is referenced in an assay that would be submitted to ENA, this file should exist within the storage of ENA.

Reviewer 2: ```suggestion
  * Assert that files referenced within the ISA-JSON metadata are uploaded to the target repositories. For example, if a FASTQ file is referenced in an assay that would be submitted to ENA, this file should exist within the storage of ENA. The ISA-JSON metadata payload should contain the checksums associated to the data files (this is requirement for SRA based repositories)
```
